src/main/java/io/github/opencubicchunks/cubicchunks/mixin/core/common/MixinIForgeDimension.java

```diff
@@ -1,19 +1,27 @@
 package io.github.opencubicchunks.cubicchunks.mixin.core.common;
 
 import io.github.opencubicchunks.cubicchunks.CubicChunks;
+import io.github.opencubicchunks.cubicchunks.mixin.core.common.world.dimension.MixinDimension;
+import io.github.opencubicchunks.cubicchunks.world.dimension.ICubicDimension;
+import net.minecraft.world.dimension.Dimension;
 import net.minecraftforge.common.extensions.IForgeDimension;
 import org.spongepowered.asm.mixin.Mixin;
 import org.spongepowered.asm.mixin.Overwrite;
+import org.spongepowered.asm.mixin.Shadow;
 
 @Mixin(IForgeDimension.class)
 public interface MixinIForgeDimension {
+
+
+    @Shadow Dimension getDimension();
+
     /**
-     * @author Voronoii
+     * @author Voronoii, Cyclonit
      */
     @Overwrite(remap = false)
     default int getHeight()
     {
-        return CubicChunks.worldMAXHeight;
+        return ((ICubicDimension) getDimension()).getMaxHeight();
```

Comment:
I think the value would be better off being in World itself for performance reasons, but someone would need to benchmark it.